packages/fuels-contract/src/contract.rs

```diff
@@ -346,6 +348,7 @@ pub struct ContractCall {
     pub message_outputs: Option<Vec<Output>>,
     pub external_contracts: Vec<Bech32ContractId>,
     pub output_param: ParamType,
+    pub custom_assets: HashMap<(AssetId, Option<Bech32Address>), u64>,
```

Comment:
Why is it an `Option<Bech32Address>` and not just a `Bech32Address`?

Reply:
I think it should be `Bech32Address` with the `None` option being replaced by the user's address.

Reply:
I figured it would be neat to also have the possibility to just add any custom amount without sending it to any address. If anything is supposed to consume that asset, it can, otherwise it gets returned to the user via Output:change.
But I guess it isn't obvious enough what happens with it, is it getting burned or returned as change?